indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -730,13 +724,66 @@ private void notifyStatus(final TaskEntry entry, final TaskStatus taskStatus, St
       taskRunner.shutdown(task.getId(), reasonFormat, args);
     }
     catch (Throwable e) {
-      // If task runner shutdown fails, continue with the task shutdown routine. We'll come back and try to
-      // shut it down again later in manageInternalPostCritical, once it's removed from the "tasks" map.
+      // If task runner shutdown fails, continue with the task shutdown routine.
       log.warn(e, "TaskRunner failed to cleanup task after completion: %s", task.getId());
     }
 
     removeTaskLock(task);
     requestManagement();
+
+    // Emit event and log
+    final ServiceMetricEvent.Builder builder = ServiceMetricEvent.builder();
+    IndexTaskUtils.setTaskStatusDimensions(builder, taskStatus);
+    emitter.emit(builder.setMetric("task/run/time", taskStatus.getDuration()));
+
+    log.info(
+        "Completed task[%s] with status[%s] in [%d]ms.",
+        task.getId(), taskStatus.getStatusCode(), taskStatus.getDuration()
+    );
+
+    if (taskStatus.isSuccess()) {
+      Counters.incrementAndGetLong(totalSuccessfulTaskCount, getMetricKey(task));
+    } else {
+      Counters.incrementAndGetLong(totalFailedTaskCount, getMetricKey(task));
+    }
+  }
+
+  /**
+   * Handles updating a task's status and incrementing/decrementing the status update accounting metrics
+   */
+  private void handleStatusWrapper(
```

Comment:
Please avoid the suffix `wrapper` as it is confusing. Maybe just use `handleStatus` or `handleCompletedStatus`.